### PR TITLE
run RichardsModel on GPU

### PR DESCRIPTION
**Pull request by juliasloan25**
## Purpose 
closes https://github.com/CliMA/ClimaLand.jl/issues/541

## To-do
- [x] add column run on GPU to buildkite (richards_comparison.jl)
- [x] add sphere run on GPU to buildkite (richards_runoff.jl)

## Current status
- column run works
  - no plots - doesn't verify accuracy, just that it runs
- sphere run works
  - no plots - doesn't verify accuracy, just that it runs
  - out of memory error fixed by increasing cache save frequency from `dt` to `100dt`

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: experiments/standalone/Soil/richards_comparison.jl [hunk 1]
```diff
@@ -100,8 +113,8 @@ clay_datapath, sand_datapath =
         # Check that simulation still has correct float type
         @assert eltype(sol.u[end].soil) == FT
 
-        # Check results and plot for Float64 simulation
-        if FT == Float64
+        # Check results and plot for Float64 simulation on CPU
+        if FT == Float64 && context.device isa ClimaComms.CPUSingleThreaded
```
Comment (juliasloan25) [inline experiments/standalone/Soil/richards_comparison.jl:117]: only plot on CPU (fails on GPU)

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -84,10 +93,21 @@ steps:
 
       - label: "Richards Runoff CPU"
         command: "julia --color=yes --project=.buildkite experiments/standalone/Soil/richards_runoff.jl"
-        artifact_paths: "experiments/standalone/Soil/artifacts/*png"
+        artifact_paths: "experiments/standalone/Soil/artifacts/cpu/*png"
```
Comment (juliasloan25) [inline .buildkite/pipeline.yml:96]: I changed the artifact paths to use the architecture type. Even though we aren't producing artifacts on GPU currently, I think this is a useful separation

**Review by kmdeck — COMMENTED** [review]
looks good, thank you!

One thought - do we want to instead run richards runoff on GPU as part of our performance pipeline?

**Review by kmdeck — APPROVED** [review]

**Comment by juliasloan25** [comment]
> One thought - do we want to instead run richards runoff on GPU as part of our performance pipeline?

Good point! Maybe we can add it in both places - so the regular CI run will make sure no PR breaks this change, even if the benchmarks pipeline isn't explicitly run, and the benchmarks pipeline wll let us monitor performance changes

**Pull request merged**